### ISSUE TITLE
Class '\PHPPgAdmin\Database\-3' not found

### ISSUE THREAD

**Issue by stephanleons** [HuasoFoundries/phppgadmin#30]
hello

using PHP 7 or 7.1 I am getting errors such as the one below, I guess it is not normal and unexpected to call the given class. Any suggestions on this perhaps?

thank you

Slim Application Error

The application could not run because of the following error:
Details
Type: Error
Message: Class '\PHPPgAdmin\Database\-3' not found
File: /www/phppgadmin/src/classes/Misc.php
Line: 146
Trace

```
#0 /www/phppgadmin/index.php(36): PHPPgAdmin\Misc->getDatabaseAccessor()
#1 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#2 /www/phppgadmin/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(41): call_user_func(Object(Closure), Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#3 /www/phppgadmin/vendor/slim/slim/Slim/Route.php(344): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#4 /www/phppgadmin/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(116): Slim\Route->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#5 /www/phppgadmin/vendor/slim/slim/Slim/Route.php(316): Slim\Route->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#6 /www/phppgadmin/vendor/slim/slim/Slim/App.php(438): Slim\Route->run(Object(Slim\Http\Request), Object(Slim\Http\Response))
#7 /www/phppgadmin/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(116): Slim\App->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#8 /www/phppgadmin/vendor/slim/slim/Slim/App.php(332): Slim\App->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#9 /www/phppgadmin/vendor/slim/slim/Slim/App.php(293): Slim\App->process(Object(Slim\Http\Request), Object(Slim\Http\Response))
#10 /www/phppgadmin/index.php(137): Slim\App->run()
#11 {main}
```

**Comment by amenadiel** [comment]
Hi.

I saw this error too. It seems that after an unsuccesful attempt to login (which means, no database info was ever retrieved), the software still tried to get a database accesor without having PostgreSQL version info to pick the right driver.

Long story short, I replaced the error handler with another one that instead throws exceptions, and I was able to capture the exception of an unsuccessful login, and display just the login screen again with a brief error message.

This can be seen in release beta.10

(I'm still not sure about what other implications may arise from the fact that I changed the error handler, but we'll see)
